apprise/plugins/workflows.py

```diff
@@ -0,0 +1,559 @@
+        self.include_image = True if (
+            include_image if include_image is not None else
+            self.template_args['image']['default']) else False
+
+        # Wrap Text
+        self.wrap = True if (
+            wrap if wrap is not None else
+            self.template_args['wrap']['default']) else False
+
+        # Our template object is just an AppriseAttachment object
+        self.template = AppriseAttachment(asset=self.asset)
+        if template:
+            # Add our definition to our template
+            self.template.add(template)
+            # Enforce maximum file size
+            self.template[0].max_file_size = self.max_workflows_template_size
+
+        # Prepare Version
+        self.api_version = version if version is not None \
+            else self.template_args['ver']['default']
+
+        # Template functionality
+        self.tokens = {}
+        if isinstance(tokens, dict):
+            self.tokens.update(tokens)
+
+        elif tokens:
+            msg = 'The specified Workflows Template Tokens ' \
+                  '({}) are not identified as a dictionary.'.format(tokens)
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        # else:  NoneType - this is okay
+        return
+
+    def gen_payload(self, body, title='', notify_type=NotifyType.INFO,
+                    **kwargs):
+        """
+        This function generates our payload whether it be the generic one
+        Apprise generates by default, or one provided by a specified
+        external template.
+        """
+
+        # Acquire our to-be footer icon if configured to do so
+        image_url = None if not self.include_image \
+            else self.image_url(notify_type)
+
+        body_content = []
+        if image_url:
+            body_content.append({
+                "type": "Image",
+                "url": image_url,
+                "height": "32px",
+                "altText": notify_type,
+            })
+
+        if title:
+            body_content.append({
+                "type": "TextBlock",
+                "text": f'**{title}**',
+                "style": "heading",
+            })
+
+        body_content.append({
+            "type": "TextBlock",
+            "text": body,
+            "style": "default",
+            "wrap": self.wrap,
+        })
+
+        if not self.template:
+            # By default we use a generic working payload if there was
+            # no template specified
+            schema = "http://adaptivecards.io/schemas/adaptive-card.json"
+            payload = {
+                "type": "message",
+                "attachments": [
+                    {
+                        "contentType":
+                        "application/vnd.microsoft.card.adaptive",
+                        "contentUrl": None,
+                        "content": {
+                            "$schema": schema,
+                            "type": "AdaptiveCard",
```

Comment:
...realised I only made the comment on the test, so it will need to go here as well:
```suggestion
                            "type": "AdaptiveCard",
                            "msteams": { "width": "full" },
```